app/src/main/java/com/example/crosses_zeros/ui/presentation/gameboard/GameBoardViewModel.kt

```diff
@@ -0,0 +1,192 @@
+    private val dataStore: AppDataStore
+) : ViewModel() {
+    private val emptyList = listOf(
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+        SignState.EMPTY,
+    )
+    private val mutableGameBoardState: MutableStateFlow<List<SignState>> =
+        MutableStateFlow(emptyList)
+    private val gameEnd: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    private val crossedOutCells: MutableStateFlow<List<Int>> = MutableStateFlow(emptyList())
+    private var response: MutableStateFlow<Int> = MutableStateFlow(0)
+
+    init {
+        viewModelScope.launch {
+            if (checkConnectionUseCase.invoke()) {
+                response.emit(getErrorResponseUseCase.invoke())
+            } else {
+                response.emit(404)
+            }
+        }
+    }
+
+    val gameState: StateFlow<GameBoardState> =
+        combine(
+            gameEnd,
+            crossedOutCells,
+            dataStore.readName,
+            response,
+            mutableGameBoardState
+        ) { gameEnd,
+            crossedOutCells,
+            lastUrl,
+            response,
+            stepsList ->
+            GameBoardState.Data(
+                cellList = stepsList,
+                gameEnd = gameEnd,
+                crossedOutCells = crossedOutCells,
+                response = response,
+                url = lastUrl,
+            )
+        }.stateIn(
+            started = SharingStarted.Lazily,
+            scope = viewModelScope,
+            initialValue = GameBoardState.Initial.Loading
+        )
+
+    fun onChangeSign(elementNumber: Int) {
+        if (mutableGameBoardState.value[elementNumber] == SignState.EMPTY ||
+            mutableGameBoardState.value[elementNumber] == element
+        ) {
+            mutableGameBoardState.update {
+                val signStates = it.toMutableList()
+                signStates[elementNumber] = element
+                signStates.toList()
+            }
+            prepareSign()
+        }
+    }
+
+    private var element: SignState = SignState.X
+    private fun prepareSign() {
+        checkWin()
+        element = if (element == SignState.X) SignState.O else SignState.X
+    }
+
+    private fun checkWin() {
+        when {
+            firstRow() || secondRow() || thirdRow() ||
+                    firstColumn() || secondColumn() || thirdColumn() ||
+                    firstDiagonal() || secondDiagonal() -> {
+                viewModelScope.launch {
+                    gameEnd.emit(true)
+                }
+                toDrawWinLine()
+            }
+        }
+    }
+
+    private fun setListData(first: Int, second: Int, third: Int) {
+        crossedOutCells.update {
+            val list = mutableListOf(first, second, third)
+            list.toList()
+        }
+    }
+
+    private fun toDrawWinLine() {
```

Comment:
This may be just getter inside state model

Reply:
it was replace